tasks/main.yml

```diff
@@ -1,11 +1,5 @@
-- name: Set version specific variables
-  include_vars: "{{ item }}"
-  with_first_found:
-    - "{{ ansible_distribution }}-{{ ansible_distribution_version }}.yml"
-    - "{{ ansible_distribution }}-{{ ansible_distribution_major_version }}.yml"
-    - "{{ ansible_distribution }}.yml"
-    - "{{ ansible_os_family }}.yml"
-    - "default.yml"
+- name: Ensure ansible_facts and variables used by role
+  include_tasks: tasks/set_vars.yml
```

Comment:
Hmm, so `include_role:` requires explicit `public: true` to expose its variables. But [include_tasks](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/include_tasks_module.html) does not have such a flag at all. Looks oddly asymmetric, but apparently works.

Just thinking aloud, nothing to change here.

Reply:
> Hmm, so `include_role:` requires explicit `public: true` to expose its variables. But [include_tasks](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/include_tasks_module.html) does not have such a flag at all. Looks oddly asymmetric, but apparently works.

A task file cannot create variables that are private to the file.  A role can create variables (in vars/) that are private to the role.

> 
> Just thinking aloud, nothing to change here.